From 121d27306e26eac0018e8676f76b825474af63ce Mon Sep 17 00:00:00 2001
From: Vallish Vaidyeshwara <vallish@amazon.com>
Date: Mon, 12 Feb 2018 22:51:01 +0000
Subject: drivers/amazon: introduce AMAZON_ENA_ETHERNET

This option is for out-of-tree ENA driver.

Signed-off-by: Munehisa Kamata <kamatam@amazon.com>
Reviewed-by: Cristian Gafton <gafton@amazon.com>
Reviewed-by: Guru Anbalagane <guruanb@amazon.com>
CR: https://cr.amazon.com/r/6834621/

Reviewed-by: Eduardo Valentin <eduval@amazon.com>
Reviewed-by: Anchal Agarwal <anchalag@amazon.com>
Signed-off-by: Vallish Vaidyeshwara <vallish@amazon.com>
CR: https://cr.amazon.com/r/8244644/
---
 drivers/amazon/Kconfig      | 9 +++++++++
 drivers/amazon/net/Makefile | 1 +
 2 files changed, 10 insertions(+)

diff --git a/drivers/amazon/Kconfig b/drivers/amazon/Kconfig
index 7cc44c84699e..eb0f5450bb1d 100644
--- a/drivers/amazon/Kconfig
+++ b/drivers/amazon/Kconfig
@@ -12,4 +12,13 @@ config AMAZON_DRIVER_UPDATES
 
 if AMAZON_DRIVER_UPDATES
 
+config AMAZON_ENA_ETHERNET
+	tristate "Elastic Network Adapter (ENA) support"
+	depends on PCI_MSI && !ENA_ETHERNET
+	---help---
+	  This driver supports Elastic Network Adapter (ENA)
+
+	  To compile this driver as a module, choose M here.
+	  The module will be called ena.
+
 endif # AMAZON_DRIVER_UPDATES
diff --git a/drivers/amazon/net/Makefile b/drivers/amazon/net/Makefile
index 6b4996dcbe52..d59ce86b1311 100644
--- a/drivers/amazon/net/Makefile
+++ b/drivers/amazon/net/Makefile
@@ -1,3 +1,4 @@
 #
 # Amazon Driver Updates
 #
+obj-$(CONFIG_AMAZON_ENA_ETHERNET)	+= ena/
-- 
2.32.0

